src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/CSharpReferencesTests.cs

```diff
@@ -108,6 +108,28 @@ public void RemoveProjectConvertsProjectReferencesBack()
             }
         }
 
+        [WpfFact]
+        [WorkItem(461967, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/461967")]
+        [WorkItem(727173, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/727173")]
+        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
+        public void AddingMetadataReferenceToProjectThatCannotCompileInTheIdeKeepsMetadataReference()
+        {
+            using (var environment = new TestEnvironment())
+            {
+                var project1 = CreateCSharpCPSProject(environment, "project1", commandLineArguments: @"/out:c:\project1.dll");
+                var project2 = CreateNonCompilableProject(environment, "project2", @"C:\project2.fsproj");
+                project2.BinOutputPath = "c:\\project2.dll";
+
+                project1.AddMetadataReference(project2.BinOutputPath, MetadataReferenceProperties.Assembly);
+
+                // We should not have converted that to a project reference, because we would have no way to produce the compilation
+                Assert.Empty(environment.Workspace.CurrentSolution.GetProject(project1.Id).AllProjectReferences);
+
+                project2.Dispose();
+                project1.Dispose();
```

Comment:
❔ These currently are not disposed if the test fails. Should they be initialized in `using` statements?

Reply:
Done of the rest of the tests do; I'm not sure if it's needed at this point or not.